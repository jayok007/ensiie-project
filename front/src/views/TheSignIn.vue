<template>
  <v-content>
    <v-container fluid>
      <v-layout align-center justify-center>
        <v-flex xs12 sm8 md6 lg4>
          <v-card class="elevation-12">
            <v-toolbar dark color="primary">
              <v-toolbar-title>Formulaire de connexion</v-toolbar-title>
              <v-spacer></v-spacer>
            </v-toolbar>

            <v-card-text>
              <v-form v-model="isValid" ref="login" lazy-validation>
                <v-text-field
                  v-model="login"
                  prepend-icon="person"
                  name="login"
                  label="Pseudo ou adresse email"
                  type="text"
                  :rules="rules"
                  @keyup.enter="submit"
                ></v-text-field>

                <v-text-field
                  v-model="password"
                  id="password"
                  prepend-icon="lock"
                  name="password"
                  label="Mot de passe"
                  type="password"
                  :rules="rules"
                  @keyup.enter="submit"
                ></v-text-field>
              </v-form>
            </v-card-text>

            <v-card-actions>
              <v-spacer></v-spacer>
              <v-btn
                color="secondary"
                @click="submit"
                :disabled="!isValid"
              >
                Connexion
              </v-btn>
            </v-card-actions>
          </v-card>
        </v-flex>
      </v-layout>
    </v-container>
  </v-content>
</template>

<script>
import Toaster from '@/toaster.js'
import { mapGetters } from 'vuex'

export default {
  name: 'TheSignIn',

  props: {
    source: String
  },

  data () {
    return {
      drawer: null,
      isValid: false,
      login: '',
      password: '',
      rules: [
        value => value.length >= 3 || 'Min 3 caractères.',
        value => value.length < 25 || 'Max 25 caractères.',
        value => !!value || 'Obligatoire.'
      ]
    }
  },

  computed: mapGetters(['isAdmin']),

  methods: {
    async submit () {
      if (!this.$refs.login.validate()) return

      try {
        await this.$store.dispatch('login', {
          login: this.login,
          password: this.password
        })

        if (this.isAdmin) {
          return this.$router.push('/admin')
        }
        this.$router.push('/feed')
      } catch (err) {
        this.$log.error(err)

        Toaster.$emit('error', 'Login ou Mot de passe incorrect')
      }
    }
  }
}
</script>
