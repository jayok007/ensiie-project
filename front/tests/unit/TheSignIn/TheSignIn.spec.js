import { shallowMount } from '@vue/test-utils'
import TheSignIn from '@/views/TheSignIn'

describe('TheSignIn', () => {
  it('should render properly', () => {
    const wrapper = shallowMount(TheSignIn)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
