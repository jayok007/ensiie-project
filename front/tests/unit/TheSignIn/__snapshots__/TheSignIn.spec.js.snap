// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSignIn should render properly 1`] = `
<vcontent-stub tag="main">
  <vcontainer-stub tag="div" fluid="">
    <vlayout-stub tag="div" align-center="" justify-center="">
      <vflex-stub tag="div" xs12="" sm8="" md6="" lg4="">
        <vcard-stub tag="div" class="elevation-12">
          <vtoolbar-stub scrollthreshold="300" dark="" color="primary">
            <vtoolbartitle-stub>Formulaire de connexion</vtoolbartitle-stub>
            <vspacer-stub></vspacer-stub>
          </vtoolbar-stub>
          <vcardtext-stub>
            <vform-stub lazyvalidation="true">
              <vtextfield-stub prepend-icon="person" name="login" label="Pseudo ou adresse email" type="text" rules="function (value) {
        return value.length >= 3 || 'Min 3 caractères.';
      },function (value) {
        return value.length < 25 || 'Max 25 caractères.';
      },function (value) {
        return !!value || 'Obligatoire.';
      }"></vtextfield-stub>
              <vtextfield-stub id="password" prepend-icon="lock" name="password" label="Mot de passe" type="password" rules="function (value) {
        return value.length >= 3 || 'Min 3 caractères.';
      },function (value) {
        return value.length < 25 || 'Max 25 caractères.';
      },function (value) {
        return !!value || 'Obligatoire.';
      }"></vtextfield-stub>
            </vform-stub>
          </vcardtext-stub>
          <vcardactions-stub>
            <vspacer-stub></vspacer-stub>
            <vbtn-stub activeclass="v-btn--active" tag="button" type="button" color="secondary" disabled="true">
              Connexion
            </vbtn-stub>
          </vcardactions-stub>
        </vcard-stub>
      </vflex-stub>
    </vlayout-stub>
  </vcontainer-stub>
</vcontent-stub>
`;
